File: src/main/java/com/angorasix/projects/core/domain/Attribute.java
package com.angorasix.projects.core.domain;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Value Object.
 * 
 * <p>Parametrized attribute that can be compared
 * 
 * @author rozagerardo
 *
 */
@Getter
@AllArgsConstructor
public final class Attribute<P> {

  private final P value;
  private final List<Attribute<?>> refs;
  private final AttributeSpecifics specifics;

}

File: src/main/kotlin/com/angorasix/projects/core/domain/Attribute.kt
package com.angorasix.projects.core.domain

/**
 * Value Object.
 *
 * Parametrized attribute that can be compared
 *
 * @author rozagerardo
 */
data class Attribute<P>(val value: P?,
                        val refs: List<Attribute<*>>?,
                        val specifics: AttributeSpecifics?) {}